### ISSUE TITLE
Corner case failures on 'general/branding/path'

### ISSUE THREAD

**Issue by GawainLynch** [bolt/bolt#1833]
As reported by moleculeFFF on IRC.

Use Case
------------

Creation of the following route will break access to http://example.com/bolt but **not** http://example.com/bolt/ (note trailing slash)
```
template:
  path:               /{template}
  defaults:            {_controller: 'Bolt\Controllers\Frontend::template'} 
```

References
--------------

This is discussed in detail on the [Silex issue tracker](https://github.com/silexphp/Silex/issues/149)

Bolt currently uses our own URL matcher written by @jcracknell in #900

**Comment by bobdenotter** [comment]
ping @GawainLynch, @rossriley or @GDmac 

Ok, since the three of you seem to know what this is about, please self-assign this to either one of you. :-)

**Comment by GawainLynch** [comment]
Assigned, but it is not an easy fix...  Will require patience and :beers:

**Comment by bobdenotter** [comment]
Is this fixed by @carsonf in #3085?

**Comment by CarsonF** [comment]
Yep yep! :beers: for all!
